herbert_core/utilities/maths/sawtooth_iarray.m

```diff
@@ -1,38 +1,47 @@
 function ivout = sawtooth_iarray (n)
-% Create array of successive runs [1,2,3..n(1),1,2,3...n(2),...]'
+% Create column vector [1;2;3..n(1); 1;2;3...n(2); ...] from integer array n
 %
 %   >> ivout = sawtooth_iarray (n)
 %
+% This is a 'sawtooth' array as each element n(i) defines a right angle triangle
+% with length n and values 1,2,3,...n(i).
+%
+%
 % Input:
 % ------
-%   n       List of length of each sawtooth section (integer)
-%          (Zeros correspond to zero length section i.e. they are
+%   n       Array with the lengths of each sawtooth section
```

Comment:
The array you create requires n to elucidate its structure. Having this as a class would make sense as then you can embed n into it. It would also make n(i)==0 sensible as you can pad unused rows. Otherwise banning n(i)==0 may be more sensible,

Reply:
I don't really understand this comment. The output is a column vector, so all rows have length one; the section for n(i) has size [n(i),1]; n(i)=0 results in a section with size [0,1] which is equivalent to being ignored. This behaviour is consistent with replicate_iarray (v, n) where n(i) results in a section of the output column vector with n(i) repeats of the value v(i) i.e. size [n(i),1]. In the case of n(i) = 0 the is a section of size [0,1].

The result is that replicate_iarray(v,n) and sawtooth_iarray(n) can be used together as they always produce a column vector with the same length = sum(n(:)), as is done in @PixelDataBase, and cases of n(i)=0 for one or more i are handled as required by some multifit commands. I've added a comment that the size of the output is [sum(n(:)), 1]

I think defining a class just adds complication and performance overheads to what is a straightforward function.